holoviews/ipython/__init__.py

```diff
@@ -85,18 +87,53 @@ def load_notebook():
 
 Parser.namespace = {'np':np, 'Cycle':Cycle, 'Palette': Palette}
 
-_loaded = False
-def load_ipython_extension(ip):
+class load_notebook_fn(param.ParameterizedFunction):
+    """
+    Parameterized function to initialize notebook resources
+    and register magics.
+    """
 
-    global _loaded
-    if not _loaded:
-        _loaded = True
-        param_ext.load_ipython_extension(ip, verbose=False)
-        load_magics(ip)
-        OutputMagic.initialize()
-        set_display_hooks(ip)
-        load_notebook()
+    bokeh = param.Boolean(default=False, doc="""Toggle inclusion of BokehJS,
+        required to render plots using the Bokeh backend.""")
```

Comment:
How about a ``js`` parameter as an ``ObjectSelector`` to state which javascript things are loaded? Default would be 'holoviews' but the allowed options would be ``['holoviews', 'bokeh']``.